packages/test-base/src/jvmTest/kotlin/io/realm/kotlin/test/jvm/RealmTests.kt

```diff
@@ -81,12 +81,13 @@ class RealmTests {
         // Closing a Realm should also cleanup our default (internal) dispatchers.
         // The core notifier and the finalizer thread will never be closed.
         val expectedThreadCount = initialThreads.size + 1 /* core-notifier */ + if (finalizerRunning) 0 else 1
-        var counter = 5 // Wait 5 seconds for threads to settle
-        while (totalThreadCount() != expectedThreadCount && counter > 0) {
+        var counter = 10 // Wait 5 seconds for threads to settle
+        while (newThreads().any { !it.isDaemon } && counter > 0) {
             delay(1000)
             counter--
         }
-        assertEquals(expectedThreadCount, totalThreadCount(), "Unexpected thread count after closing realm: ${newThreads()}")
+        val totalThreadCount = totalThreadCount()
```

Comment:
For the sake of debugging, we should print the stack trace of all non-daemon thread to find where they're are blocking

Reply:
I thought about this, but it would massively expand the output and so far all the cases we had for this, it was enough to get the name of the thread to figure out what was going on. So for now I think I would prefer to keep as as is.